### PR TITLE
identify: only store _reported_ multiaddrs

### PR DESCRIPTION
**Pull request by Stebalien**
We still tell the remote host about the observed addr but we don't store it. That way, we give them a chance to decide if they want to actually use and advertise it.

Ideally, we'd distinguish between local information and signed routing information but we don't do that yet.

Hopefully this will prevent nodes behind relays from telling everyone else to connect to their peers through these relays.

### CODE REVIEW

**Comment by Stebalien** [comment]
Actually, this is a bit interesting... We only add this remote address if it has the same protocol sequence as an address advertised by the peer. That means we _shouldn't_ add the address here unless the peer adds the relay to their observed address set.

So, this may still may prevent the address explosion issue however, it won't fix the actual problem. To do that, we just need to remove observed relay addresses.

File: p2p/protocol/identify/id.go [hunk 1]
```diff
@@ -412,6 +409,11 @@ func (ids *IDService) consumeObservedAddress(observed []byte, c inet.Conn) {
 		return
 	}
 
+	if !HasConsistentTransport(c.RemoteMultiaddr(), ids.Host.Addrs()) {
+		log.Debugf("ignoring observed multiaddr that doesn't match the transports of any addresses we're announcing", c.RemoteMultiaddr())
+		return
+	}
```
Comment (vyzo) [inline p2p/protocol/identify/id.go]: so what happens if we are behind a NAT and advertise a relay addr, and connect to a peer through a different relay (one which we don't advertise)? It seems that we might pollute our advertised set to include this relay too.

**Comment by Stebalien** [comment]
Let's discuss in #575. I'm not sure if this is right.

**Comment by Stebalien** [comment]
Re-opened in #577 but not as a fix for #575.

**Pull request closed without merging**